components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -80,7 +104,20 @@ export default function ProjectSettingsView() {
 
     return (
         <ProjectSettingsPage project={project}>
-            <Heading2>Prebuilds</Heading2>
+            <Heading2>Project Name</Heading2>
+            <form onSubmit={updateProjectName}>
+                <TextInputField
+                    hint="The name can be up to 32 characters long."
+                    value={projectName}
+                    error={badProjectName}
+                    onChange={setProjectName}
+                />
+
+                <Button className="mt-4" htmlType="submit" disabled={project?.name === projectName || !!badProjectName}>
+                    Update Name
+                </Button>
```

Comment:
**question**: Disabled buttons are generally a cancer for UX, and we should aim to remove them all when possible, but this is reusing the pattern we have across organization and user settings. Thoughts on keeping this always enabled?

Reply:
> Disabled buttons are generally a cancer for UX
want to share why?
I have no real preference here. But we should do this consistently and in other places disabling the button is good (e.g. create workspace)

Reply:
Sounds good. Let's keep this as is for now to mimic the behavior of the organization naming and will try to update in a separate PR. 🏓